# Obtención de datos geográficos — INEGI

## Proyecto Agronomía

Esta libreta documenta la obtención y preparación de la información
geográfica utilizada en el proyecto.

Las geometrías de las entidades federativas y municipios de México
se obtienen mediante los servicios geográficos de INEGI.

La lógica de adquisición y procesamiento se encuentra implementada
en el paquete `proyecto_agronomia`, por lo que esta libreta funciona
como una interfaz reproducible para ejecutar, inspeccionar y validar
el proceso.

## Objetivo

Obtener:

- geometrías de los 32 estados de México;
- geometrías municipales;
- coordenadas representativas de cada municipio.

Estas coordenadas serán utilizadas posteriormente para consultar
las variables climáticas históricas.

## Flujo de trabajo

1. Descargar o cargar las geometrías municipales de INEGI.
2. Descargar o cargar las geometrías estatales.
3. Generar un punto representativo para cada municipio.
4. Validar claves geográficas y coordenadas.
5. Guardar el resultado para la etapa climática.

In [1]:
from proyecto_agronomia.dataset import (
    obtener_estados_inegi,
    obtener_municipios_inegi,
)

from proyecto_agronomia.features import (
    guardar_coordenadas_municipios,
)

2026-09-30 12:15:04.015 | INFO     | proyecto_agronomia.config:<module>:14 - PROJ_ROOT path is: B:\MCD2\JW\agro\proyecto_agronomia


## 1. Geometrías municipales

Se obtienen las geometrías de los municipios de México.

La función reutiliza el archivo local cuando este ya existe, evitando
realizar nuevamente la descarga.

In [2]:
municipios_geo = obtener_municipios_inegi()

municipios_geo.head()

2026-09-30 12:15:34.898 | INFO     | proyecto_agronomia.dataset:obtener_municipios_inegi:99 - Cargando municipios INEGI desde archivo local.


,cvegeo,cve_ent,cve_mun,nomgeo,cve_cab,nom_cab,pob_total,pob_femenina,pob_masculina,total_viviendas_habitadas,geometry
0,01001,01,001,Aguascalientes,0001,Aguascalientes,948990,486917,462073,266942,"MULTIPOLYGON (((-102.10732 22.07476, -102.1069..."
1,01008,01,008,San José de Gracia,0001,NaN,9552,5020,4532,2381,"MULTIPOLYGON (((-102.45537 22.31212, -102.4550..."
2,01011,01,011,San Francisco de los Romo,0001,NaN,61997,31292,30705,16391,"MULTIPOLYGON (((-102.15938 22.099, -102.15638 ..."
3,01009,01,009,Tepezalá,0001,NaN,22485,11371,11114,5425,"MULTIPOLYGON (((-102.17738 22.36243, -102.1796..."
4,01007,01,007,Rincón de Romos,0001,NaN,57369,29268,28101,13584,"MULTIPOLYGON (((-102.22684 22.37393, -102.2266..."


In [3]:
print(f"Municipios disponibles: {len(municipios_geo):,}")
print(f"CRS: {municipios_geo.crs}")
print(f"Columnas: {municipios_geo.columns.tolist()}")

Municipios disponibles: 2,478
CRS: EPSG:4326
Columnas: ['cvegeo', 'cve_ent', 'cve_mun', 'nomgeo', 'cve_cab', 'nom_cab', 'pob_total', 'pob_femenina', 'pob_masculina', 'total_viviendas_habitadas', 'geometry']


## 2. Geometrías estatales

Se obtienen las geometrías correspondientes a las 32 entidades
federativas de México.

In [4]:
estados_geo = obtener_estados_inegi()

estados_geo.head()

2026-09-30 12:16:27.221 | INFO     | proyecto_agronomia.dataset:obtener_estados_inegi:169 - Cargando estados INEGI desde archivo local.


,cvegeo,cve_ent,nomgeo,nom_abrev,pob_total,pob_femenina,pob_masculina,total_viviendas_habitadas,geometry
0,01,01,Aguascalientes,Ags.,1425607,728924,696683,386671,"MULTIPOLYGON (((-102.29739 22.45527, -102.2955..."
1,02,02,Baja California,BC,3769020,1868431,1900589,1149563,"MULTIPOLYGON (((-114.11561 28.09639, -114.1152..."
2,03,03,Baja California Sur,BCS,798447,392568,405879,240660,"MULTIPOLYGON (((-109.89451 22.87476, -109.8946..."
3,04,04,Campeche,Camp.,928363,471424,456939,260824,"MULTIPOLYGON (((-91.9565 20.197, -91.95636 20...."
4,05,05,Coahuila de Zaragoza,Coah.,3146771,1583102,1563669,901249,"MULTIPOLYGON (((-102.31549 29.88, -102.31509 2..."


In [5]:
print(f"Entidades federativas: {len(estados_geo):,}")
print(f"CRS: {estados_geo.crs}")

Entidades federativas: 32
CRS: EPSG:4326


## 3. Coordenadas representativas municipales

Para consultar posteriormente información climática es necesario
asociar cada municipio con una coordenada.

Se calcula un punto representativo dentro de cada geometría municipal
y posteriormente se transforma al sistema de coordenadas geográficas
EPSG:4326.

In [6]:
municipios_coordenadas = guardar_coordenadas_municipios(
    municipios_geo
)

municipios_coordenadas.head()

2026-09-30 12:17:30.799 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:303 - Municipios: 2,478
2026-09-30 12:17:30.800 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:306 - Claves duplicadas: 0
2026-09-30 12:17:30.800 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:309 - Latitudes faltantes: 0
2026-09-30 12:17:30.800 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:312 - Longitudes faltantes: 0
2026-09-30 12:17:30.801 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:316 - Rango latitud: 14.634916 → 32.416923
2026-09-30 12:17:30.802 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:322 - Rango longitud: -116.882750 → -86.914552
2026-09-30 12:17:30.830 | SUCCESS  | proyecto_agronomia.features:guardar_coordenadas_municipios:339 - Coordenadas municipales guardadas en B:\MCD2\JW\agro\proyecto_agronomia\data\interim\inegi\municipios_coordenadas_inegi.cs

,Idestado,Idmunicipio,cvegeo,Nommunicipio,Latitud,Longitud
0,1,1,01001,Aguascalientes,21.848487,-102.305449
1,1,8,01008,San José de Gracia,22.154297,-102.524108
2,1,11,01011,San Francisco de los Romo,22.021698,-102.251799
3,1,9,01009,Tepezalá,22.243608,-102.192713
4,1,7,01007,Rincón de Romos,22.250711,-102.348371


In [7]:
print(
    f"Municipios: {len(municipios_coordenadas):,}"
)

print(
    "Claves duplicadas:",
    municipios_coordenadas.duplicated(
        ["Idestado", "Idmunicipio"]
    ).sum(),
)

print(
    "Coordenadas faltantes:",
    municipios_coordenadas[
        ["Latitud", "Longitud"]
    ].isna().sum().sum(),
)

print(
    f"Latitud: "
    f"{municipios_coordenadas['Latitud'].min():.4f} → "
    f"{municipios_coordenadas['Latitud'].max():.4f}"
)

print(
    f"Longitud: "
    f"{municipios_coordenadas['Longitud'].min():.4f} → "
    f"{municipios_coordenadas['Longitud'].max():.4f}"
)

Municipios: 2,478
Claves duplicadas: 0
Coordenadas faltantes: 0
Latitud: 14.6349 → 32.4169
Longitud: -116.8828 → -86.9146


## Resultado

La información geográfica de INEGI quedó preparada para las siguientes
etapas del proyecto.

El conjunto de coordenadas municipales utiliza como identificadores:

- `Idestado`
- `Idmunicipio`

Estas claves permiten relacionar posteriormente la información
geográfica con los registros agrícolas.

Las coordenadas `Latitud` y `Longitud` se utilizarán para consultar
los datos climáticos históricos de cada municipio.